system-parachains/coretime-paseo/src/coretime.rs

```diff
@@ -345,6 +351,69 @@ impl pallet_broker::Config for Runtime {
 	type AdminOrigin = EnsureRoot<AccountId>;
 	type SovereignAccountOf = SovereignAccountOf;
 	type MaxAutoRenewals = ConstU32<100>;
-	type PriceAdapter = pallet_broker::MinimumPrice<Balance, MinimumEndPrice>;
+	type PriceAdapter =
+		MinimumPriceWithFixedTargetPrice<Balance, MinimumEndPrice, FixedTargetPrice>;
 	type MinimumCreditPurchase = MinimumCreditPurchase;
 }
+
+/// `AdaptPrice` like `CenterTargetPrice`, but with an extra fixed target price and fixed floor
+/// price.
+///
+/// This price adapter behaves exactly like `CenterTargetPrice`, except that it takes a minimum
+/// price and a target price and makes sure that the returned `end_price` is never lower than that.
+///
+/// Target price is fixed and never adjusted based on the previous sale performance.
+pub struct MinimumPriceWithFixedTargetPrice<Balance, MinPrice, TargetPrice>(
+	core::marker::PhantomData<(Balance, MinPrice, TargetPrice)>,
+);
+
+impl<Balance: FixedPointOperand, MinPrice: Get<Balance>, TargetPrice: Get<Balance>>
+	AdaptPrice<Balance> for MinimumPriceWithFixedTargetPrice<Balance, MinPrice, TargetPrice>
+{
+	fn leadin_factor_at(when: FixedU64) -> FixedU64 {
+		CenterTargetPrice::<Balance>::leadin_factor_at(when)
+	}
+
+	fn adapt_price(performance: SalePerformance<Balance>) -> AdaptedPrices<Balance> {
+		let mut proposal = CenterTargetPrice::<Balance>::adapt_price(performance);
+		let min_price = MinPrice::get();
+		let target_price = TargetPrice::get();
+		// Fix floor price:
+		proposal.end_price = min_price;
+		// Fix target price:
+		proposal.target_price = target_price;
+
+		proposal
```

Comment:
```suggestion
		let end_price = MinPrice::get();
		let target_price = TargetPrice::get();

		AdaptedPrices { end_price, target_price }
```

Could simplify this a bit